### PR TITLE
Bugfix/car engine at start

### PR DESCRIPTION
**Pull request by DJSteins**
Fixes car engine playing during countdown and adds some better logic for car engine sound

### CODE REVIEW

File: Assets/_PartyParkade/Script/Managers/StateManager.cs [hunk 1]
```diff
@@ -114,6 +127,7 @@ public void PlayGame()
             GameEventMessage.SendEvent(DoozyKeys.ResumeGameEvent);
             StartCoroutine(BlockInput());
             OnPlay?.Invoke();
+            PlayerManager.Players.ForEach(player => player.Car.MuteEngine(false));
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Managers/StateManager.cs:130]: The StateManager shouldn't call the PlayerManager stuff. 
The Playermanager can subscribe to the OnPlay Delegate and do this stuff. 

Same as a PreGame Delegate.

File: Assets/_PartyParkade/Script/Game/Car/CarController.cs [hunk 2]
```diff
@@ -87,6 +97,12 @@ public Vector3 GetVelocityChange(Vector3 velocity, float verticalMovement, Vecto
             // Newton's 2nd Law simple acceleration model
             float verticalAcceleration = GetVerticalAcceleration(verticalMovement);
             Vector3 acceleration = GetNewAcceleration(verticalAcceleration, GetDrag(velocity), forward);
+
+            // engine stuff
+            m_wheelSpeed = Mathf.Lerp(m_wheelSpeed, velocity.magnitude, WindDownStrength * Time.deltaTime);
+            float volume = m_wheelSpeed <= MinWheelSpeed ? MinWheelVolume : 1f;
+            m_wheelVolume = Mathf.Lerp(m_wheelVolume, volume, WindDownStrength * Time.deltaTime);
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/Car/CarController.cs:104]: Can you put this stuff inside CarAudioSystem instead?

**Comment by mikeslgoh** [comment]
[When the countdown starts car engine sound should be off](https://trello.com/c/mhPgqoXY/701-when-the-countdown-starts-car-engine-sound-should-be-off)

**Comment by DJSteins** [comment]
> Made comments
> And what did u change with the PlayerCar.Prefab?

Serialized the CarController into the CarAudioSystem

File: Assets/_PartyParkade/Script/Managers/PlayerManager.cs [hunk 3]
```diff
@@ -84,6 +84,9 @@ public void Initialize()
         }
 
         SetupThreePlayerExtraQuadrantIfNeeded();
+
+        StateManager.OnPlay += () => { SetCarEnginesMute(false); };
+        StateManager.OnPreGame += () => { SetCarEnginesMute(true); };
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Managers/PlayerManager.cs:89]: Should also unsubscribe this somewhere

Comment (DJSteins) [reply Assets/_PartyParkade/Script/Managers/PlayerManager.cs:89]: Can I unsubscribe using lambdas again, or do I need to use a reference to an actual function?

Comment (m4rtint) [reply Assets/_PartyParkade/Script/Managers/PlayerManager.cs:89]: Oh...hmm good question, i never tried that before.

Comment (DJSteins) [reply Assets/_PartyParkade/Script/Managers/PlayerManager.cs:89]: i'll keep it simple and refactor them into their own methods

**Comment by m4rtint** [comment]
Still have the Merge Conflict

**Pull request merged**